### PR TITLE
[android] jni clean up - missing a couple DeleteLocalRef

### PR DESCRIPTION
**Pull request by osana**

### CODE REVIEW

**Comment by osana** [comment]
@tobrun please look at the changes. I am not 100% sure about them - it was easier to do then discuss.

File: platform/android/src/annotation/polygon.cpp [hunk 1]
```diff
@@ -50,12 +50,18 @@ float Polygon::getOpacity(jni::JNIEnv& env, jni::Object<Polygon> polygon) {
 
 mbgl::Color Polygon::getFillColor(jni::JNIEnv& env, jni::Object<Polygon> polygon) {
     static auto field = Polygon::javaClass.GetField<int>(env, "fillColor");
-    return *conversion::convert<mbgl::Color, int>(env, polygon.Get(env, field));
+    auto jColor = polygon.Get(env, field);
+    mbgl::Color color =  *conversion::convert<mbgl::Color, int>(env, jColor);
+    jni::DeleteLocalRef(env, jColor);
```
Comment (tobrun) [inline platform/android/src/annotation/polygon.cpp]: jColor is of the type jint and these are just aliases for normal native value types, so there are no special lifetime or cleanup considerations.

File: platform/android/src/map/camera_position.cpp [hunk 2]
```diff
@@ -34,6 +34,7 @@ mbgl::CameraOptions CameraPosition::getCameraOptions(jni::JNIEnv& env, jni::Obje
     static auto zoom = CameraPosition::javaClass.GetField<jni::jdouble>(env, "zoom");
 
     auto center = LatLng::getLatLng(env, position.Get(env, target));
+    jni::DeleteLocalRef(env, target);
```
Comment (tobrun) [inline platform/android/src/map/camera_position.cpp]: great find!

File: platform/android/src/map/image.cpp [hunk 3]
```diff
@@ -16,7 +16,9 @@ mbgl::style::Image Image::getImage(jni::JNIEnv& env, jni::Object<Image> image) {
     auto width = image.Get(env, widthField);
     auto pixelRatio = image.Get(env, pixelRatioField);
     auto pixels = image.Get(env, bufferField);
-    auto name = jni::Make<std::string>(env, image.Get(env, nameField));
+    auto jName = image.Get(env, nameField);
+    auto name = jni::Make<std::string>(env, jName);
+    jni::DeleteLocalRef(env, jName);
```
Comment (tobrun) [inline platform/android/src/map/image.cpp]: great find as well, do we need to delete the pixels local ref?

Comment (osana) [reply platform/android/src/map/image.cpp]: @tobrun  Turns out that jni::GetArrayRegion(env, *pixels, 0, size, reinterpret_cast<jbyte*>(premultipliedImage.data.get()));

**Pull request merged**